### PR TITLE
feat(sensor): Adds the log trigger

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

Do not merge. For discussion.

I want a solution so that users can easily debug events. 

https://docs.google.com/document/d/1XTxHYO5Jshc5EzCxZ20Kr-vNBQLTwHWB8vT15BUJ7YU/edit

image
image

### CODE REVIEW

File: common/retry.go [hunk 1]
```diff
@@ -53,9 +53,7 @@ func GetConnectionBackoff(backoff *apicommon.Backoff) *wait.Backoff {
 	if backoff == nil {
 		return &result
 	}
-	if &backoff.Duration != nil {
```
Comment (alexec) [inline common/retry.go:56]: lint: `staticcheck`

File: controllers/sensor/validate.go [hunk 2]
```diff
@@ -514,9 +514,6 @@ func validateK8sTriggerPolicy(policy *v1alpha1.K8SResourcePolicy) error {
 	if policy.Labels == nil {
 		return errors.New("resource labels are not specified")
 	}
-	if &policy.Backoff == nil {
```
Comment (alexec) [inline controllers/sensor/validate.go:517]: lint: `staticcheck`

File: eventsources/sources/generic/validate.go [hunk 3]
```diff
@@ -6,8 +6,8 @@ import (
 )
 
 func (el *EventListener) ValidateEventSource(ctx context.Context) error {
-	if &el.GenericEventSource == nil {
-		return fmt.Errorf("event source can't be empty")
+	if el == nil {
```
Comment (alexec) [inline eventsources/sources/generic/validate.go:9]: lint: `staticcheck`

File: sensors/policy/resource-labels.go [hunk 4]
```diff
@@ -34,7 +34,7 @@ type ResourceLabels struct {
 
 func (rl *ResourceLabels) ApplyPolicy() error {
 	from := rl.Trigger.Policy.K8s.Backoff
-	if rl.Trigger.Policy.K8s == nil || rl.Trigger.Policy.K8s.Labels == nil || &from == nil {
+	if rl.Trigger.Policy.K8s == nil || rl.Trigger.Policy.K8s.Labels == nil {
```
Comment (alexec) [inline sensors/policy/resource-labels.go:37]: lint: `staticcheck`

**Comment by alexec** [comment]
Over to you @whynowy and @VaibhavPage

File: examples/sensors/log.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Sensor
+metadata:
+  name: log
+spec:
+  template:
```
Comment (whynowy) [inline examples/sensors/log.yaml:6]: Does not need a serviceAccount here.

**Review by whynowy — COMMENTED** [review]
This is a great idea for debugging!

Just a small change required.

File: examples/sensors/log.yaml (hunk 5 above)
Comment (alexec) [reply examples/sensors/log.yaml]: done

**Comment by alexec** [comment]
Ready for review.

**Comment by VaibhavPage** [comment]
> This is a great idea for debugging!
> 
> 
> 
Totally agree!!

**Review by VaibhavPage — APPROVED** [review]

**Comment by alexec** [comment]
Merged. I created a new milestone (v1.2) and added it.

**Pull request merged**